server/routes/utils/authentication.js

```diff
@@ -0,0 +1,26 @@
+const Utility = require(`${serverRoot}/lib/Utility`);
+
+const authentication = (req, res, next) => {
+  try {
+    let authCheck = false;
+    if (typeof req.session !== 'undefined') {
+      if (typeof req.session.data.signin_info !== 'undefined') {
+        if (req.session.data.signin_info.signed_in === 1) {
+          authCheck = true;
+        }
+      }
+    }
+    if (!authCheck) {
+      throw new Error('User not signed in');
+    } else {
+      next();
+    }
+  } catch (err) {
+    Utility.logException(err);
+    const port = req.socket.localPort && req.socket.localPort !== '80' ? ':' + req.socket.localPort : '';
+    const returnUrl = `${req.protocol}://${req.hostname}${port}${req.originalUrl}`;
+    return res.redirect(302, `${process.env.ACCOUNT_URL}/signin?return_to=${returnUrl}`);
+  }
+};
```

Comment:
Are we redirecting when authentication is successful?

Reply:
No there is a ! which means not.

Reply:
Ah I missed the `next()`